File: 백준/Silver/1021. 회전하는 큐/회전하는 큐.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");

const [N, M] = input[0].split(" ").map(Number);
const arr = input[1].split(" ").map(Number);
//const clothes = input.slice(2).map((_) => _.trim().split(" "));

function main() {
  const queue = Array.from({ length: N }, (v, i) => i + 1);
  let answer = 0;
  arr.forEach((el) => {
    if (queue.indexOf(el) < queue.length / 2) {
      //왼쪽으로 이동
      while (true) {
        const item = queue.shift();
        if (item === el) {
          break;
        }
        queue.push(item);
        answer++;
      }
    } else {
      //오른쪽으로 이동
      while (true) {
        const item = queue.pop();
        if (item === el) {
          answer++;
          break;
        }
        queue.unshift(item);
        answer++;
      }
    }
  });

  console.log(answer);
}

main();

File: 백준/Silver/1021. 회전하는 큐/회전하는 큐.cc
#include <iostream>
#include <vector>
#include<deque>
using namespace std;

int findIndex(deque<int> queue, int item) {
    for(int i=0;i<queue.size(); i++) {
        if(item == queue[i]) {
            return i;
        }
    }   
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

   // freopen("input.txt", "r", stdin);

    int N, M;
    cin >> N >> M;

    int arr[M];
    for(int i=0; i<M; i++) {
       cin >> arr[i];
    }

    deque<int> queue(N);
    for(int i=0; i<N; i++) {
       queue[i] = i+1;
    }

    int answer = 0;

    for(int el : arr) {
        if(findIndex(queue, el) <= queue.size() / 2) {
            //왼쪽으로 이동
            while(true) {
                int item = queue.front();
                queue.pop_front();
                if(item == el) {
                    break;
                }
                queue.push_back(item);
                answer++;
            }
        }
        else {
            //오른쪽으로 이동
            while(true) {
                int item = queue.back();
                queue.pop_back();
                if(item == el) {
                    answer++;
                    break;
                }
                queue.push_front(item);
                answer++;
            }
        }
    }

    cout << answer;

    

    return 0;
}